packages/react/src/Details/Details.tsx

```diff
@@ -14,7 +18,63 @@ const Details = styled.details<SxProp>`
   ${sx};
 `
 
-Details.displayName = 'Details'
+export const StyledSummary = styled.summary<SxProp>`
+  ${sx};
+`
+
+const Root = React.forwardRef<HTMLDetailsElement, ComponentProps<typeof StyledDetails>>(
+  ({children, ...props}: ComponentProps<typeof StyledDetails>, forwardRef) => {
+    const detailsRef = React.useRef<HTMLDetailsElement>(null)
+    const ref = useMergedRefs(forwardRef, detailsRef)
+    const [hasSummary, setHasSummary] = useState(false)
+
+    useEffect(() => {
+      const {current: details} = detailsRef
+      if (!details) {
+        return
+      }
+      const summary = details.querySelector('summary:not([data-default-summary])')
+      setHasSummary(!!summary)
+      warning(!summary, 'No summary found, a default summary will be rendered instead.')
+    }, [children])
```

Comment:
ended up landing here with the DOM querySelector approach, let me know if you have any thoughts @joshblack

Reply:
@francinelucca one thing that came to mind with `useEffect` used this way with `children` is that it will re-run this on each render since the identity of `children` will change (unless memoized). So this would be equivalent to `useEffect` without a dependency array.

Separately, one thing I was confused by is using a `warning` here. Wanted to double check, do we want to support a fallback summary or do we want to force someone to provide a summary?

Reply:
thanks for pointing that out, I did not know that about children "changing" on every render! using `MutationObserver` on https://github.com/primer/react/pull/5015/commits/e98319968785de44cfe9e8b167977fe6cfe72097

For the warning, I am adding the default summary if `!hasSummary` on line 63, I just figured I'd add a warning to alert of the fact that a summary wasn't supplied to the component, in case that wasn't intended. But if you think that's confusing/misguiding I can remove 👀
<img width="804" alt="image" src="https://github.com/user-attachments/assets/e42c3cc0-a470-40de-ba1a-a4421bd5c444">